### PR TITLE
(maint) Clarify upgrade process in docs

### PR DESCRIPTION
**Pull request by MikaelSmith**
Briefly describes upgrade process, and points to Puppet docs for more
details. Also added more detailed description of the settings that are
changed during upgrade.

Removed an empty, unused class to clean up documentation.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -51,13 +52,22 @@ Add the class to agents you want to upgrade:
 include ::puppet_agent
 ~~~
 
+It will install the latest released version of Puppet from Puppet Collection 1.
+
+The recommended way to upgrade with this module is to first standup a Puppet Server 2.1 master - which supports backward compatibility with Puppet 3 agents - and point the agent to upgrade at it. Once you've confirmed the agent runs successfully against the new master, and thus the Puppet 4 language, apply the class to the agent and confirm that it checks back in after a successful upgrade. Further details on upgrading are available in Puppet's documentation.
```
Comment (MikaelSmith) [inline README.markdown]: Further details I'm mainly referring to https://github.com/puppetlabs/puppet-docs/blob/july2015_improved_upgrade/source/puppet/4.2/reference/upgrade_major_agent.markdown, which is still unreleased. The eventual URL for that will probably be linked from http://docs.puppetlabs.com/puppet/latest/reference/#upgrading-from-puppet-3x, but I'm not sure of the final link so I'm wary to put it in the module.

**Comment by MikaelSmith** [comment]
@trevor-vaughan does this help?

@jbondpdx could you give my doc changes a review?

**Comment by trevor-vaughan** [comment]
:+1: This makes things 100% clear

**Comment by MikaelSmith** [comment]
Added this in response to https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/puppet-dev/wcI8-409UI4.

**Comment by jbondpdx** [comment]
Looks good! I'd like to see us link out to the docs, but agree it's better to avoid it if we don't know the correct link yet. 

I'm suggesting a few small changes for style and clarity:

This installs the latest released version of Puppet from Puppet Collection 1.

To upgrade with this module, first stand up a Puppet Server 2.1 master---which supports backward compatibility with Puppet 3 agents---and point the agent you want to upgrade at that master.

**Comment by MikaelSmith** [comment]
Updated.

**Comment by MikaelSmith** [comment]
Upgrade guide went live yesterday at http://docs.puppetlabs.com/puppet/4.2/reference/upgrade_major_pre.html, so updating with that link.

File: README.markdown [hunk 2]
```diff
@@ -51,13 +51,22 @@ Add the class to agents you want to upgrade:
 include ::puppet_agent
 ~~~
 
+This installs the latest released version of Puppet from Puppet Collection 1.
+
+To upgrade with this module, first stand up a Puppet Server 2.1 master---which supports backward compatibility with Puppet 3 agents---and point the agent you want to upgrade at that master. Once you've confirmed the agent runs successfully against the new master, and thus the Puppet 4 language, apply the class to the agent and confirm that it checks back in after a successful upgrade. Further details on upgrading are available [here](http://docs.puppetlabs.com/puppet/4.2/reference/upgrade_major_pre.html).
```
Comment (MikaelSmith) [inline README.markdown]: @jbondpdx @gguillotte I leaned towards fixing to a version rather than using latest; also not sure about what to call the link.

**Comment by jbondpdx** [comment]
:+1:

**Comment by gguillotte** [comment]
:+1:

**Pull request merged**